admin/aioseop_module_class.php

```diff
@@ -1737,15 +1737,12 @@ function admin_enqueue_styles( $hook_suffix ) {
 				wp_enqueue_style( 'aioseop-module-style-rtl', AIOSEOP_PLUGIN_URL . 'css/modules/aioseop_module-rtl.css', array( 'aioseop-module-style' ), AIOSEOP_VERSION );
 			}
 
-			// Uses WP Scripts to load the current platform version of jQuery UI CSS .
-			$wp_scripts = wp_scripts();
-			wp_enqueue_style(
-				'aioseop-jquery-ui-css',
-				'https://ajax.googleapis.com/ajax/libs/jqueryui/' . $wp_scripts->registered['jquery-ui-core']->ver . '/themes/smoothness/jquery-ui.min.css',
-				false,
-				AIOSEOP_VERSION,
-				false
-			);
+			// Uses WP Scripts to load the current platform version of jQuery UI CSS.
+			// Loads only if it is not already loaded in order to not cause a conflict between 2 different versions.
+			if ( ! wp_style_is( 'jquery-ui', 'registered' ) && ! wp_style_is( 'jquery-ui', 'enqueued' ) ) {
+				$wp_scripts = wp_scripts();
+				wp_enqueue_style( 'jquery-ui', '//ajax.googleapis.com/ajax/libs/jqueryui/' . $wp_scripts->registered['jquery-ui-core']->ver . '/themes/smoothness/jquery-ui.min.css', false, AIOSEOP_VERSION, false );
```

Comment:
I believe `jquery-ui` needs to be prefixed to prevent conflict with other plugins. I've had experiences in the past where other plugins use the same id to load a different file/version. Either ours before theirs, or theirs before ours.

The line should also be broken down to multiple lines (like it was); especially for long lines like this.

Reply:
I don't agree that we should be in a position to load 2 different CSS for jquery-ui. That's why the check for whether the style has been loaded or not. Unless you have an example I can take a look at to understand this better.

Reply:
Loading 2 jQuery UI is still going to happen regardless when plugins use a different name/id. I had also thought about loading 2, but I've found it has been better to load the version we intend to use.

Also, using a generic name is better left with WordPress; which would be the most likely scenario. So, WP would have priority in that case.

Plus, this PR isn't intended to fix an issue like that. So, it's a bit odd why this was even changed when there is no reported issue. The IF check is good, but it's name/id that has me concerned.

Reply:
The PR didn't "fix" it. I made my changes first and when I synced your changes also appeared. So I decided to use the correct method. I would go with the generic naming than a plugin specific name unless you have a specific real-life example of a problem that this can cause.

Reply:
Ok, I see where you made changes.

It's been awhile, but here on some of the issues I had tracked...

https://github.com/Advanced-Post-List/advanced-post-list/issues/26
https://github.com/Advanced-Post-List/advanced-post-list/issues/44
https://github.com/Advanced-Post-List/advanced-post-list/issues/74

1) Multiple versions are still loaded, regardless.
2) If there is a different script loaded before aioseop's with the same id, it will prevent ours.
3) Generic names are better left with WP, and even WP says to "prefix everything".
https://www.sitepoint.com/enqueuing-scripts-styles-wordpress/
https://make.wordpress.org/core/2018/12/06/javascript-packages-and-interoperability-in-5-0-and-beyond/
https://developer.wordpress.org/plugins/plugin-basics/best-practices/#avoid-naming-collisions
https://wpgurus.net/enqueue-scripts-style-sheets-on-shortcode-pages/
4) The most likely scenario is WP enqueueing the script; if and when it does get added. All other plugins are expected to prefix everything.
5) When debugging the script, it would help to prefix it to ensure we are enqueueing and can check if it did enqueue in WP_Scripts.

Reply:
That is your plugin so I was looking for behavior that someone else has implemented just to understand how others go about it. Your implementation will not differ from your suggestion :)

We have 2 alternatives:

If we prefix:
- we don't have to use the `wp_style_is`
- 2 different CSS can be loaded. How will that affect the UI?

If we don't prefix
- we will retain `wp_style_is`
- if someone else has loaded another CSS theme different from ours with the generic prefix, it will be loaded. How will that affect the UI?

I can't find any source that says one method is better over the other. It would have been easier to decide.

@michaeltorbert will need your inputs here.

Reply:
You asked for real life examples, and I thought you wanted my past examples. 

I still added 4 other links, 2 of them are from WP. Which basically reiterates what I've been saying.

> For vendor scripts in particular, there is a chance for conflict if a plugin or theme registers their own copy of the dependency with the same un-prefixed name. There is a greater likelihood if there is a large version mismatch between the registered scripts.
>
> In general, a plugin should seek to avoid naming collisions by following the best practice of “Prefix Everything” as prescribed in the Plugin Handbook. While this can sometimes result in multiple copies of a script being present in a page, it reduces the chance for breakage and grants WordPress the option to use the unprefixed name in the future.

https://make.wordpress.org/core/2018/12/06/javascript-packages-and-interoperability-in-5-0-and-beyond/

Which then links to this https://developer.wordpress.org/plugins/plugin-basics/best-practices/#avoid-naming-collisions.

Reply:
Makes sense. I've made this change.